pkg/i2gw/providers/common/converter.go

```diff
@@ -319,12 +319,28 @@ func (rg *ingressRuleGroup) configureBackendRef(paths []ingressPath) ([]gatewayv
 	var errors field.ErrorList
 	var backendRefs []gatewayv1beta1.HTTPBackendRef
 
+	// track unique backends to avoid duplicates
+	uniqueBackends := make(map[string]interface{})
+
 	for i, path := range paths {
 		backendRef, err := toBackendRef(path.path.Backend, field.NewPath("paths", "backends").Index(i))
 		if err != nil {
 			errors = append(errors, err)
 			continue
 		}
+
+		if path.path.Backend.Service != nil && backendRef.BackendObjectReference.Port != nil {
+			backend := fmt.Sprintf("%s/%d", backendRef.BackendObjectReference.Name, *backendRef.BackendObjectReference.Port)
+
+			// skip duplicates
+			if _, exists := uniqueBackends[backend]; exists {
+				continue
+			}
+
+			// add new unique backend
+			uniqueBackends[backend] = nil
+		}
```

Comment:
I tried with this:

```go
func (rg *ingressRuleGroup) configureBackendRef(paths []ingressPath) ([]gatewayv1beta1.HTTPBackendRef, field.ErrorList) {
	var errors field.ErrorList
	var backendRefs []gatewayv1beta1.HTTPBackendRef

	// track unique backends to avoid duplicates
	uniqueBackendRefs := make(map[*gatewayv1beta1.HTTPBackendRef]interface{})
	for i, path := range paths {
		backendRef, err := toBackendRef(path.path.Backend, field.NewPath("paths", "backends").Index(i))
		if err != nil {
			errors = append(errors, err)
			continue
		}

		if _, exists := uniqueBackendRefs[&gatewayv1beta1.HTTPBackendRef{BackendRef: *backendRef}]; exists {
			continue
		}
		uniqueBackendRefs[&gatewayv1beta1.HTTPBackendRef{BackendRef: *backendRef}] = nil

		backendRefs = append(backendRefs, gatewayv1beta1.HTTPBackendRef{BackendRef: *backendRef})
	}

	return backendRefs, errors
}
```

But just checking for duplication does not work as the backendref has pointers. So even if the value of the ports are same, it fails to be identified as a duplicate because the pointers are different.

So I think we should create a new key for the uniquebackendrefs that has all the values needed from the backendref. But it might be difficult to maintain in the future where we might have to update the fields as backendrefs's spec changes.

What do you suggest we do here?

Reply:
you are right, nice catch ! :)

This is what I came up with. (indeed if the spec change we will need to change but i dont anticipate it will change that often)

Also I a different solution -mine is probably not that good

```go
func (rg *ingressRuleGroup) configureBackendRef(paths []ingressPath) ([]gatewayv1beta1.HTTPBackendRef, field.ErrorList) {
	var errors field.ErrorList
	type BackendRefMapKey struct {
		Name      gatewayv1beta1.ObjectName
		Namespace gatewayv1beta1.Namespace
		Port      gatewayv1beta1.PortNumber
		Group     gatewayv1beta1.Group
		Kind      gatewayv1beta1.Kind
	}

	mapKey := func(backendRef gatewayv1beta1.BackendRef) BackendRefMapKey {
		if backendRef.Port != nil {
			return BackendRefMapKey{
				Name: backendRef.Name,
				Port: *backendRef.Port,
			}
		}
		return BackendRefMapKey{
			Name:  backendRef.Name,
			Group: *backendRef.Group,
			Kind:  *backendRef.Kind,
		}
	}

	uniqueBackendRefs := map[BackendRefMapKey]gatewayv1beta1.BackendRef{}

	for i, path := range paths {
		backendRef, err := toBackendRef(path.path.Backend, field.NewPath("paths", "backends").Index(i))
		if err != nil {
			errors = append(errors, err)
			continue
		}
		uniqueBackendRefs[mapKey(*backendRef)] = *backendRef
	}
	var backendRefs []gatewayv1beta1.HTTPBackendRef
	for _, br := range uniqueBackendRefs {
		backendRefs = append(backendRefs, gatewayv1beta1.HTTPBackendRef{BackendRef: br})
	}
	return backendRefs, errors
```

Reply:
Okay, if the spec does not change, this is the solution. I could also think of a similar solution. I will push the changes.

Reply:
awesome, please also add a test for duplicated backend refs